ledgereth/comms.py

```diff
@@ -81,31 +82,13 @@ def get_with_data(
         return cmd.encode()
 
 
-def translate_exception(exp: Exception) -> Exception:
-    """Translate a CommException into something more useful to the user"""
-    string_err = str(exp)
-    if "No dongle found" in string_err:
-        return Exception("Ledger Nano not found")
-    elif "6804" in string_err:
-        return Exception("Ledger appears to be locked?")
-    elif any(x in string_err for x in ["6700", "6d00"]):
-        return Exception("Please open the Ethereum app on your Ledger device")
-    elif "6a80" in string_err:
-        return Exception(
-            "General failure: Invalid transaction, blind signing not allowed in settings, or"
-            " something else?"
-        )
-    else:
-        return exp
-
-
 def dongle_send(dongle, command_string: str) -> bytes:
     """Send a command to the dongle"""
     hex_command = LedgerCommands.get(command_string)
     try:
         return dongle.exchange(hex_command)
     except CommException as err:
-        raise translate_exception(err)
+        raise LedgerError.transalate_comm_exception(err)
```

Comment:
```suggestion
        raise LedgerError.transalate_comm_exception(err) from err
```

to preserve the exception stack?

Reply:
I'm not familiar with this, can you point me towards the docs?  Not sure how I'd look that up.

Reply:
Found it.  https://docs.python.org/3/library/exceptions.html#exception-context